### ISSUE TITLE
editor.codeActionsOnSave.source.fixAll doesn't work when specified for multiple languages

### ISSUE THREAD

**Issue by tao-cumplido** [microsoft/vscode#140082]
Issue Type: <b>Bug</b>

When `editor.codeActionsOnSave.source.fixAll` is specified in a combined setting for multiple languages it won't work, e.g.

```json
{
  "[javascript][typescript]": {
    "editor.codeActionsOnSave": {
      "source.fixAll.eslint": true
    }
  }
}
```

does not work while

```json
{
  "[javascript]": {
    "editor.codeActionsOnSave": {
      "source.fixAll.eslint": true
    }
  },
  "[typescript]": {
    "editor.codeActionsOnSave": {
      "source.fixAll.eslint": true
    }
  }
}
```

does.

The `editor.formatOnSave` setting doesn't seem to be affected and works fine either way.

VS Code version: Code - Insiders 1.64.0-insider (Universal) (990ad4044032b05c0ebf0f67f800c5845baad4c7, 2022-01-04T05:14:46.209Z)
OS version: Darwin arm64 20.6.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|4, 3, 3|
|Memory (System)|16.00GB (0.06GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.2.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv695:30137379
vsins829:30139715
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30258533
pythonvspyt551:30291412
pythonptprofiler:30281269
vshan820:30294714
pythondataviewer:30285072
vscod805:30301674
pythonvspyt200:30323110
bridge0708:30335490
bridge0723:30353136
vsaa593cf:30376535
pythonvs932:30404738
vscop804:30404766
vscop453:30404998
vsclayoutctrc:30405799
py55gd98:30408621
vscexrecpromp3c:30407757
azactmsalcf:30409288

```

</details>

**Comment by mjbvz** [comment]
@sandy081 Is there anything our code needs to do to support the new multiple language settings?

**Comment by sandy081** [comment]
I do not think you have to do anything, it shall work. User says the same works for `editor.formatOnSave`. Please check how you are reading `editor.codeActionsOnSave` setting ?

**Comment by mjbvz** [comment]
@dbaeumer I debugged through this and think eslint is the root cause. When I changed the setting to:

```json
  "[javascript][typescript]": {
        "editor.codeActionsOnSave": {
            "source.organizeImports": true
        }
    }
```

I see the organize imports action applying properly. However using `"source.fixAll.eslint": true`, the request made to extensions [here](https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/codeEditor/browser/saveParticipants.ts#L363) returns an empty array of actions
